Source/SessionManager.swift

```diff
@@ -328,25 +362,18 @@ open class SessionManager {
 
     // MARK: Private - Download Implementation
 
-    private func download(
-        _ downloadable: Downloadable,
-        to destination: Request.DownloadFileDestination)
-        -> Request
-    {
+    private func download(_ downloadable: Downloadable, to destination: Request.DownloadFileDestination) -> Request {
         var downloadTask: URLSessionDownloadTask!
 
         switch downloadable {
-        case .request(let request):
-            queue.sync {
-                downloadTask = self.session.downloadTask(with: request)
-            }
-        case .resumeData(let resumeData):
-            queue.sync {
-                downloadTask = self.session.downloadTask(withResumeData: resumeData)
-            }
+        case let .request(urlRequest):
+            let urlRequest = adapt(urlRequest)
```

Comment:
Yeah...there's definitely room for improvement, which is why I reworked it in a following PR! The `adapt` API is now an extension on `URLRequest` which takes the `adapter` as an argument. That cleaned up things a bunch. I'd say let's move this through as is right now and we can revisit in a downstream PR if you still think it's a bit clunky.